src/Service.Tests/CosmosTests/SamplerTests.cs

```diff
@@ -76,72 +76,74 @@ await executor
+        /// Verifies the functionality of the <c>TopNExtractor</c> class with various configurations to ensure it accurately samples the top N records.
         /// </summary>
         /// <param name="count">The maximum number of top records to retrieve.</param>
         /// <param name="maxDays">The maximum number of days to filter records. If null, no day-based filtering is applied.</param>
         /// <param name="expectedCount">The expected number of records returned by the sampler.</param>
         /// <remarks>
-        /// This test case ensures that the <c>TopNSampler</c> behaves as expected under different sampling scenarios, including scenarios where
+        /// This test case ensures that the <c>TopNExtractor</c> behaves as expected under different sampling scenarios, including scenarios where
         /// maximum days are not specified. It checks that the sampling logic correctly handles the specified count and optionally applies
         /// date-based filtering based on the <c>maxDays</c> parameter.
         /// </remarks>
-        [TestMethod(displayName: "TopNSampler Scenarios")]
-        [DataRow(1, 0, 1, DisplayName = "TopNSampler: Retrieve 1 record when max days are not specified.")]
-        [DataRow(5, null, 5, DisplayName = "TopNSampler: Retrieve 5 records when max days are null")]
-        [DataRow(5, 2, 3, DisplayName = "TopNSampler: Retrieve 3 records with max days configured as 2.")]
-        public async Task TestTopNSampler(int count, int? maxDays, int expectedCount)
+        [TestMethod(displayName: "TopNExtractor Scenarios")]
+        [DataRow(1, 0, 1, DisplayName = "TopNExtractor: Retrieve 1 record when max days are not specified.")]
+        [DataRow(5, null, 5, DisplayName = "TopNExtractor: Retrieve 5 records when max days are null")]
+        [DataRow(5, 2, 3, DisplayName = "TopNExtractor: Retrieve 3 records with max days configured as 2.")]
+        public async Task TestTopNExtractor(int count, int? maxDays, int expectedCount)
         {
-            Mock<TopNSampler> topNSampler = new(_containerWithIdPk, count, maxDays, _mockLogger.Object);
+            Mock<TopNExtractor> topNExtractor = new(_containerWithIdPk, count, maxDays, _mockLogger.Object);
 
             if (maxDays is null || maxDays == 0)
             {
-                maxDays = 10;
+                maxDays = TopNExtractor.MAX_DAYS;
             }
 
-            topNSampler
+            topNExtractor
                 .Setup<long>(x => x.GetTimeStampThreshold())
                 .Returns((long)(_sortedTimespansIdPk[0] - maxDays));
 
-            List<JsonDocument> result = await topNSampler.Object.GetSampleAsync();
-            Assert.AreEqual(expectedCount, result.Count);
+            List<JsonDocument> result = await topNExtractor.Object.GetSampleAsync();
+
+            // The actual count can vary by 1 record because of time based record creation
```

Comment:
what does this mean? example? are records created with same time stamp?

Reply:
If we create records without a delay, they will all have the same timestamp, which would prevent us from testing time-based filtering in test cases. To address this, I've added a delay to ensure that each record has a unique timestamp.

Reply:
Because we're relying on a delay to create records with different timestamps, I noticed that, in some cases, the actual result intermittently varies by one record, particularly in pipelines. To ensure these tests aren't flaky, I had to adjust the assertion.